mitodl/mit-open
# Use pluggy to trigger search index updates

~Based on PR #159 which should be merged first~

# What are the relevant tickets?
Closes #108 

# Description (What does it do?)
- Modifies the code to have search index updates triggered via pluggy plugins


# How can this be tested?
- Set `OCW_*` and `XPRO_`settings in your .env to be the same as in RC
- Run `./manage.py backpopulate_ocw_data --course-name 21w-747-rhetoric-spring-2005`
- Wait a minute or so then go to `http://localhost:8063/api/v1/learning_resources_search/?q=rhetoric`.  That ocw course should show up in results.
- Go to `http://localhost:8063/api/v1/content_file_search/?q=%22abbreviated%20in%20the%20schedule%20outlined%22`, you should get a result back.
- Run `./manage.py backpopulate_ocw_data --delete --course-name 21w-747-rhetoric-spring-2005`
- Wait a minute or so then go to `http://localhost:8063/api/v1/learning_resources_search/?q=rhetoric`.  That ocw course should not show up in results
- Go to `http://localhost:8063/api/v1/content_file_search/?q=%22abbreviated%20in%20the%20schedule%20outlined%22` again, you should get 0 results
- Run `./manage.py backpopulate_xpro_data`
- Wait a minute then go to `http://localhost:8063/api/v1/learning_resources_search/?platform=xpro`.  You should get some results.
- Run `./manage.py backpopulate_xpro_data --delete`
- Wait a minute then go to `http://localhost:8063/api/v1/learning_resources_search/?platform=xpro`.  You should get 0 results.
